pandas/tests/extension_arrays/test_json.py

```diff
@@ -0,0 +1,115 @@
+import collections
+import itertools
+import numbers
+import operator
+import random
+import string
+import sys
+
+import numpy as np
+import pytest
+
+
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.arrays import ExtensionArray
+
+from .base import BaseArrayTests, BaseDtypeTests
+
+
+class JSONDtype(ExtensionDtype):
+    type = collections.Mapping
+    name = 'json'
+
+    @classmethod
+    def construct_from_string(cls, string):
+        if string == cls.name:
+            return cls()
+        else:
+            raise TypeError("Cannot construct a '{}' from "
+                            "'{}'".format(cls, string))
+
+
+class JSONArray(ExtensionArray):
+    dtype = JSONDtype()
+
+    def __init__(self, values):
+        for val in values:
+            if not isinstance(val, collections.Mapping):
+                raise TypeError
+        self.data = values
+
+    def __getitem__(self, item):
+        # TDOO: fancy indexing
+        if isinstance(item, numbers.Integral):
+            return self.data[item]
+        elif isinstance(item, np.ndarray) and item.dtype == 'bool':
+            return type(self)([x for x, m in zip(self, item) if m])
+        elif isinstance(item, collections.Sequence):
+            return type(self)([self.data[i] for i in item])
```

Comment:
At the moment (I think) this is not in our requirements / specification (although numpy arrays do this as well). Was this needed for test to get passed? (i.e. do we use this somewhere internally, instead of using `take` ?)

Anyway, I would either add to specs, or remove it here to test that it is not required.

Reply:
Sorry, I accidentally relied on this for the test

```python
    def test_take_sequence(self, data):
        result = pd.Series(data[[0, 1, 3]])
 ```

I don't think pandas uses it anywhere. Will rewrite that test.